jax/_src/scipy/special.py

```diff
@@ -1901,3 +1902,22 @@ def hyp1f1(a, b, x):
   lambda b_dot, primal_out, a, b, x: _hyp1f1_b_derivative(a, b, x) * b_dot,
   lambda x_dot, primal_out, a, b, x: _hyp1f1_x_derivative(a, b, x) * x_dot
 )
+
+@_wraps(osp_special.comb, module='scipy.special')
+def comb(N: Array, k: Array, exact: bool =False, repetition: bool =False):
```

Comment:
Input arrays should be annotated `ArrayLike`, and we should annotate the return value as ` -> Array`.